### DISCUSSION TITLE
SVG imported through Webpack resourceQuery/next-image-loader returns undefined src asset prefix

### DISCUSSION THREAD

**Discussion by jeroenroumen-acc** [vercel/next.js#30868] [Help]
### What version of Next.js are you using?

12.0.8

### What version of Node.js are you using?

12.22.8

### What browser are you using?

Chrome, Firefox

### What operating system are you using?

macOS

### How are you deploying your application?

next dev

### Describe the Bug

Currently on Next.js 11.3 the project has a slightly extended webpack config in the `next.config.js` to accomodate for importing SVGs through SVGR (`v6.2.0`). On Next.js 11 this works pretty well.

Using resourceQueries to specify which loader is using and with some specific options like so:

```
webpack(config) {
    config.module.rules.push({
      test: /\.svg$/,
      oneOf: [
        {
          resourceQuery: /image/, // example.svg?image
          use: ['next-image-loader'],
        },
        {
          resourceQuery: /icon/, // example.svg?icon
          use: [
            {
              loader: '@svgr/webpack',
              options: {
                icon: true,
              },
            },
          ],
        },
        {
          use: ['@svgr/webpack'],
        },
      ],
    });

    return config;
  },
```

One difference in the `next-image-loader` between Next 11 and 12 I noticed was this:
https://github.com/vercel/next.js/blob/v11.1.3/packages/next/build/webpack/loaders/next-image-loader.js#L20
https://github.com/vercel/next.js/blob/v12.0.8/packages/next/build/webpack/loaders/next-image-loader.js

```
# Next 11.3
const outputPath = '/_next' + interpolatedName

# Next 12.0.8
const outputPath = assetPrefix + '/_next' + interpolatedName
```

assetPrefix actually turns out to be undefined but as a string. This results in the path down below where _next is prefixed with undefined (`undefined/_next/static/...`). So it seems somewhere between the webpack resourceQuery and the `next-image-`loader` this value when grabbed is undefined and prepended as a string.

` const { isServer, isDev, assetPrefix, basePath } = this.getOptions()`

here: https://github.com/vercel/next.js/blob/v12.0.8/packages/next/build/webpack/loaders/next-image-loader.js#L11

I tried adding an empty string to the `assetPrefix` in the `next.config.js` but that doesn't seem to resolve it unfortunately.

```Server Error
Error: Failed to parse src "undefined/_next/static/media/graphic.c9879546.svg" on `next/image`, if using relative image it must start with a leading slash "/" or be an absolute URL (http:// or https://)```

### Expected Behavior

When importing an svg with a resourceQuery while using the above mentioned Webpack config to return a `StaticImageData` object instead of an error.

import exampleSVG from '../public/images/example.svg?image'

### To Reproduce

1. Create a blank Next.js 12.0.8 application with `npx create-next app <app-name>`.
2. Install the SVGR library (latest version) `npm install --dev @svgr/webpack`
3. Use above mentioned Webpack config with resourceQueries and the SVGR library.
4. Import an svg `import exampleSVG from '../public/images/example.svg?image'` like this
5. Start the application by doing a `next dev`

**Comment by timneutkens** [comment]
Converted to discussions help section as you're customizing the webpack configuration so this is not a bug in Next.js.

**Reply by jeroenroumen-acc** [reply]
Yes my bad, sorry for that! Definitely should be a discussion.

**Comment by jeroenroumen-acc** [comment]
Updated the problem description since we came back around to updating our application to Next 12 and still having this issue. I believe the problem is a combination of the webpack setup with resourceQueries and the updated `next-image-loader`.

**Reply by ShaySong99** [reply]
The same problem when i update to next@12. 
It seems to solve it by add assetPrefix in the example. I have'nt check carefully whther there would be any other problem.

```
config.module.rules.push({
			test: /\.svg$/i,
			issuer: /\.[jt]sx?$/,
			resourceQuery: /icon/,
			use: ['@svgr/webpack'],
		});
		config.module.rules.push({
			test: /\.svg$/i,
			issuer: /\.[jt]sx?$/,
			resourceQuery: { not: [/icon/] },
			loader: 'next-image-loader',
			options: { assetPrefix: '' },
		});
```

**Comment by 1aron** [comment]
Here is my path configuration to make `next-image-loader` exclude `inlineSvg` resource queries:
```js
const nextImageRule = config.module.rules.find((rule) => rule.loader === 'next-image-loader')

if (nextImageRule) {
    nextImageRule.resourceQuery.not.push(/inlineSvg/)
}

config.module.rules.push({
    resourceQuery: /inlineSvg/,
    use: [
        {
            loader: '@svgr/webpack',
            options: {
                svgoConfig: {
                    plugins: [],
                },
            },
        },
    ],
})
```
Now you can:
```jsx
import MySVG from './my.svg?inlineSvg';

<MySVG />

// or

import Image from 'next/image';
import mySvg from './my.svg';

<Image src={mySvg} ... />
```
